App/Components/CreateThreadModal.tsx

```diff
@@ -53,8 +69,19 @@ class CreateThreadComponent extends React.Component<
       }
       this.setState({ submitted: true })
       this.props.completeScreen(this.state.value)
+      const threadConfig = {
+        type:
+          this.props.type !== undefined ? this.props.type : Thread.Type.OPEN,
+        sharing:
+          this.props.sharing !== undefined
+            ? this.props.sharing
+            : Thread.Sharing.SHARED,
```

Comment:
couldn't this just be `this.props.sharing || Thread.Sharing.SHARED`

Reply:
It used to be this, actually, but when `this.props.sharing` is equal to `Thread.Sharing.NOT_SHARED`, which is the enum value 0, it evaluates to false and the logic breaks down, so I had to explicitly check it against undefined.

Reply:
ah hahah! aaron was just talking about this issue this morning. good call.

Reply:
i meant to say, 'Good, you passed my test'.

Reply:
haha yeah it's a really pernicious bug! took me forever to figure out why it was selectively failing haha.